core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -235,6 +236,8 @@
 
     private Optional<TableExecuteHandle> tableExecuteHandle = Optional.empty();
 
+    // names of tables and aliased relations. All parts of a QualifiedName are lower-cased.
```

Comment:
Why are they lower cased? What happens with delimited names? That would seem to go against what the SQL spec describes.

Reply:
The comment is not precise. It should say "All names are resolved case-insensitive".

I am aware that this is not compliant with the SQL spec. Regarding resolution of co-partitioned tables, I chose to follow how table references are resolved in Trino.

Currently in Trino, all table references, including CTEs and aliased relations, are resolved case-insensitive, and sadly, the 'delimited' property is disregarded. In theory, we could do spec-compliant resolution for the co-partitioned tables, but I think it would be unintuitive to the user. Similarly, we decided to resolve TF names in the usual non-ANSI way to avoid confusion between different function types.

If we decided to resolve co-partitioned table references in the spec-compliant way, we'd have to decide how "permissive" we want to be with respect to schema and catalog names.